src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -37,6 +37,17 @@ namespace winrt::TerminalApp::implementation
 
         _MakeTabViewItem();
         _CreateContextMenu();
+
+        // Add an event handler for the header control to tell us when they want their title to change
+        _headerControl.HeaderTitleChanged([weakThis = get_weak()](auto&& title) {

```

Comment:
Do we need to author a custom event (`HeaderTitleChanged`) for this? Since `Title` is observable, shouldn't we be able to do this with the automatic `PropertyChanged` event?

For ex, TerminalPage.cppL1143-1163, we'd probably do something similar:
```c++

_headerControl.PropertyChanged([weakThis=get_weak()](auto&&, const WUX::Data::PropertyChangedEventArgs& args) {
    if (auto tab{ weakThis.get() })
    {
        if (args.PropertyName() == L"Title")
        {
            tab->SetTabText(title);
        }
    }
});
```

Reply:
Ah I should rename this event - the purpose of this event is to let the hosting tab know that the user tried to change the title to x, and once the tab receives that event it tells the headerControl to change its title to x.

The reason for this is so its always the tab that has ownership over the title